src/main/java/mil/dds/anet/views/ViewResponseFilter.java

```diff
@@ -24,14 +39,53 @@ public ViewResponseFilter(AnetConfiguration config) {
 
   @Override
   public void filter(ContainerRequestContext requestContext,
-      ContainerResponseContext responseContext) throws IOException {
+      ContainerResponseContext responseContext) {
     // Don't cache requests other than GET, and don't cache selected media types
     final MediaType mediaType = responseContext.getMediaType();
     if (!HttpMethod.GET.equals(requestContext.getMethod()) || mediaType == null
         || uncachedMediaTypes.stream().anyMatch(mt -> mt.equals(mediaType))) {
       responseContext.getHeaders().put(HttpHeaders.CACHE_CONTROL,
           ImmutableList.of("no-store, no-cache, must-revalidate, post-check=0, pre-check=0"));
       responseContext.getHeaders().put(HttpHeaders.PRAGMA, ImmutableList.of("no-cache"));
+      // Log only requests other than GET and if userPrincipal exists
+      if (!HttpMethod.GET.equals(requestContext.getMethod())
+          && requestContext.getSecurityContext().getUserPrincipal() != null) {
+        // This code snippet resolves the situation that caused empty user information to be
+        // written to the access.log file
+        // Start
+        Principal userPrincipal = new Principal() {
+          @Override
+          public String getName() {
+            final SecurityContext secContext = requestContext.getSecurityContext();
+            Principal p = secContext.getUserPrincipal();
+            return p.getName();
+          }
+        };
+        UserIdentity userId = new DefaultUserIdentity(null, userPrincipal, null);
+        Request baseRequest = Request.getBaseRequest(request);
+        baseRequest.setAuthentication(new UserAuthentication(null, userId));
+        // End
+
+        // Store user activities in Person bean
+        if (requestContext.getSecurityContext().getUserPrincipal() instanceof Person) {
+          final Person person = (Person) requestContext.getSecurityContext().getUserPrincipal();
+          AnetObjectEngine.getInstance().getPersonDao().logActivitiesByDomainUsername(
+              person.getDomainUsername(), new HashMap<String, String>() {
+                {
+                  put("ip", request.getRemoteAddr() != null ? request.getRemoteAddr() : "-");
```

Comment:
I have a slight preference for:
```
                  put("ip", request.getRemoteAddr() == null ? "-" : request.getRemoteAddr());
```
to avoid the "dangling else" problem.